### ISSUE TITLE
external-dns chart improvements

### ISSUE THREAD

**Issue by ubergesundheit** [giantswarm/roadmap#367]
- Check if the external-dns chart can add the required annotation on AWS
- Revisit sidecar + init container for checking if role is available
- Make `external-dns` exit non-zero on `records retrieval failed: failed to list hosted zones`

**Comment by JosephSalisbury** [comment]
@ubergesundheit what do you think about not doing this, and moving to the upstream chart instead?

**Comment by ubergesundheit** [comment]
Unsure if upstream chart does these things currently, if yes then lets do it

**Comment by mcharriere** [comment]
- Check if the external-dns chart can add the required annotation on AWS

This is part of https://github.com/giantswarm/roadmap/issues/411 and it should be solved in v3

- Revisit sidecar + init container for checking if role is available

There is a ticket already for this: https://github.com/giantswarm/giantswarm/issues/26851 which we can work on already.

-  Make external-dns exit non-zero on records retrieval failed: failed to list hosted zones

@ubergesundheit what is the reasoning here? did we have some problems in the past that were solved by restarting? I'm just collecting info in case we want to push this to upstream.

**Comment by ubergesundheit** [comment]
> did we have some problems in the past that were solved by restarting?

yes, kiam related. Restarting solved external dns unable to list zones.

**Comment by mcharriere** [comment]
> yes, kiam related. Restarting solved external dns unable to list zones.

Ah yes, that makes sense. now that you mentioned, I've seen the same problem with IRSA. Ok cool. let's see if failing-fast is possible.

**Comment by weatherhog** [comment]
closing this issue
